kogito-codegen/src/main/java/org/kie/kogito/codegen/JSonSchemaGenerator.java

```
@@ -0,0 +1,111 @@
+        builder.forFields().withIgnoreCheck(this::isNotUserTaskParam);
+        SchemaGenerator generator = new SchemaGenerator(builder.build());
+        ObjectWriter writer = new ObjectMapper().writer();
+        Map<String, List<Class<?>>> map = stream.filter(this::isUserTaskClass).collect(Collectors.groupingBy(this::getKey));
+        Collection<GeneratedFile> files = new ArrayList<>();
+        for (Map.Entry<String, List<Class<?>>> entry : map.entrySet()) {
+            ObjectNode merged = null;
+            for (Class<?> c : entry.getValue()) {
+                ObjectNode read = generator.generateSchema(c);
+                if (merged == null)
+                    merged = read;
+                else
+                    merged.setAll(read);
+            }
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            writer.writeValue(outputStream, merged);
+            files.add(new GeneratedFile(Type.JSON_SCHEMA, entry.getKey() + ".json", outputStream.toByteArray()));
+        }
+        return files;
+    }
+
+
+    private String getKey(Class<?> c) {
+        return c.getAnnotation(UserTask.class).taskName();
+    }
+
+    private boolean isUserTaskClass(Class<?> c) {
+        return c.isAnnotationPresent(UserTask.class);
+    }
+
+    private boolean isNotUserTaskParam(FieldScope fieldScope) {
+        return !fieldScope.getDeclaringType().getErasedType().isAnnotationPresent(UserTask.class) || fieldScope.getAnnotation(UserTaskParam.class) == null;
+    }
+
+    public static final Stream<Class<?>> getClassStream(Path parentPath, ClassLoader cl) throws IOException {
+        return Files.walk(parentPath).filter(p -> p.toString().endsWith(".class")).map(parentPath::relativize).map(p -> {
+            try {
+                return cl.loadClass(getClassNameFromPath(p));
+            } catch (ClassNotFoundException e) {
+                return (Class<?>) null;
```

Is this actually possible?
I think we should throw an exception in this case instead of mute the error.
Wdyt?

---

Yes, you are right, a class file should contain a class that can be uploaded by the passed classloader. 
In inicial implementation I was using a custom class loader that potentially might not found some of the classe, but this can be safely removed now.

---

Hmmm, after a second thought, even if there is a mistmatch between the classloader and the path provider, I think is safer to silently ignore this (consequence being not schema is generated) rather than breaking the whole generation procedure. More opinions on that are welcome.

---

Is this phase considered as optional? 
What about add a property to `GenerateJsonSchemaMojo` like failsafe with `true` as default for now and propagate this option to `JSonSchemaGenerator`?
It could be overkill so up to you :)
In any case please add some logging information (warning or error level)

---

Yes, this is smarter, I will do that

---

Exception is now thrown from the lambda. 
All exceptions are catched by Mojo and rethrown as MavenFailure (not MavenExecution), so it can be ignored using proper flags when invoking maven